### PR TITLE
feat: github action to install cargo-tarpaulin 

### PR DESCRIPTION
**Pull request by jvanz**
## Description

his action downloads latest stable release of cargo-tarpaulin and installs
that inside of the GitHub action path.

Fix https://github.com/kubewarden/kubewarden-controller/issues/667

### CODE REVIEW

**Comment by viccuad** [comment]
Thanks for tackling this.

Agree with Flavio. Maybe a solution is to use `cargo binstall`, which does the work of checking crates.io, and going to the GH releases to fetch the provided binaries.

They provide a minimal GH action, that we can just copy-paste into our actions, see https://github.com/cargo-bins/cargo-binstall?tab=readme-ov-file#in-github-actions.

**Comment by flavio** [comment]
We have to download the tarball from the tarpaulin release page and put the `cargo-tarpaulin` binary into the $PATH. We can copy&paste https://github.com/kubewarden/github-actions/tree/main/crane-installer and adapt it.

Once the `cargo-tarpaulin` is into the $PATH, `cargo` will automatically find it when `cargo tarpaulin` is invoked

File: tarpaulin-install/action.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: 'tarpaulin-installer'
+description: 'Install tarpaulin and includes it inside of PATH'
+branding:
+  icon: 'package'
+  color: 'blue'
+inputs:
+  VERSION:
+    description: 'tarpaulin release to be installed'
+    required: false
+    default: "0.27.3"
```
Comment (flavio) [inline tarpaulin-install/action.yml:10]: In this case we have the current latest version hard-coded. I would drop this input variable and use the code snippet from below

File: tarpaulin-install/action.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  steps:
+    - shell: bash
+      run: |
+        #!/bin/bash
+        set -e
+
+        INSTALL_DIR=$HOME/.tarpaulin
+
+        mkdir -p $INSTALL_DIR
+
```
Comment (flavio) [inline tarpaulin-install/action.yml:22]:
```bash
LATEST_RELEASE=$(curl -s https://api.github.com/repos/xd009642/tarpaulin/releases/latest | jq -r '.tag_name')
```

this currently assigns `0.27.3` to `LATEST_RELEASE`

**Review by flavio — CHANGES_REQUESTED** [review]

**Review by viccuad — APPROVED** [review]
LGTM. Maybe worth it to do the bump to 3.1.18 already in this PR, with
`sed --in-place 's/3\.1\.17/3\.1\.18/' $(find . -maxdepth 3 -name "*.yml")`.

**Pull request merged**